### PR TITLE
Various misc. linalg tests

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/linalg/symmetric.jl [hunk 1]
```diff
@@ -296,10 +297,14 @@ end
...
             @testset "mat * vec" begin
                 @test Symmetric(asym)*x+y ≈ asym*x+y
+                xadj = x.'
+                @test xadj * Symmetric(asym).' ≈ xadj * asym
                 @test x' * Symmetric(asym) ≈ x' * asym
 
                 @test Hermitian(aherm)*x+y ≈ aherm*x+y
+                xadj = x'
                 @test x' * Hermitian(aherm) ≈ x' * aherm
+                @test xadj * Hermitian(aherm)' ≈ xadj * aherm
```
Comment (Sacha0) [inline test/linalg/symmetric.jl]: This test seems to check only that `adjoint(Hermitian(aherm))` is approximately `Hermitian(aherm)`?

File: test/linalg/symmetric.jl [hunk 2]
```diff
@@ -296,10 +297,14 @@ end
         @testset "linalg binary ops" begin
             @testset "mat * vec" begin
                 @test Symmetric(asym)*x+y ≈ asym*x+y
+                xadj = x.'
+                @test xadj * Symmetric(asym).' ≈ xadj * asym
```
Comment (Sacha0) [inline test/linalg/symmetric.jl]: Beyond the immediately following test, this test seems to check only that `transpose(Symmetric(asym))` is approximately `Symmetric(asym)`?

**Review by Sacha0 — COMMENTED** [review]
Is the purpose of the `convert` tests simply to check the associated entry points, which are otherwise (implementation) covered by the immediately adjacent `AbstracyArray(...)` tests?

File: test/linalg/symmetric.jl (hunk 1 above)
Comment (kshyatt) [reply test/linalg/symmetric.jl]: I'm after testing [these fallbacks](https://codecov.io/gh/JuliaLang/julia/src/master/base/linalg/symmetric.jl#L318) - will this not do it?

**Comment by kshyatt** [comment]
Re: `convert` - yeah, exactly

File: test/linalg/symmetric.jl (hunk 1 above)
Comment (Sacha0) [reply test/linalg/symmetric.jl]: If this test exercises the `*(A::Hermitian, x::Bool)` fallback, I miss how? Also, if you substitute `x'` for `xadj` in this test, the result is `x' * Hermitian(aherm)' ≈ x' * aherm`, which differs from the immediately preceding test only in the `'` on `Hermitian(aherm)'`? Am I missing something? :)

**Comment by Sacha0** [comment]
> Re: convert - yeah, exactly

Cheers then :). Calling `convert(Array, ...)` should test each method in the associated `convert` chain ending with `convert(AbstractMatrix, ...)`, so you could simply replace the `AbtractArray(...)` tests with the `convert(Array, ...)` tests you have here. (And if that doesn't hold for any of these `convert` chains, perhaps we should fix that.)

File: test/linalg/symmetric.jl (hunk 1 above)
Comment (fredrikekre) [reply test/linalg/symmetric.jl]: > If this test exercises the *(A::Hermitian, x::Bool) fallback, I miss how?

That's what I thought first too, but the link points to L318, but my browser does not end up there...

Comment (Sacha0) [reply test/linalg/symmetric.jl]: That makes much more sense! Thank you! :)

Comment (Sacha0) [reply test/linalg/symmetric.jl]: Alright, I see the intent now :). Perhaps a comment or two describing the intent would be worthwhile? Perhaps something along the lines of `@test (xadj = x'; xadj * Hermitian(aherm)' ≈ xadj * aherm) # exercise A_mul_Bc(RowVector, Hermitian)`?

Comment (kshyatt) [reply test/linalg/symmetric.jl]: can do :) sorry about the weird linking issues

Comment (Sacha0) [reply test/linalg/symmetric.jl]: Not your fault! :)

**Comment by kshyatt** [comment]
@Sacha0 good to go?

**Comment by Sacha0** [comment]
Thanks `@kshyatt`! :)

**Pull request merged**